From 87390c421c28e90adda96bacd587565912aff3be Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Fri, 14 Dec 2018 09:16:43 +0000
Subject: [PATCH] Point to new mirror as kodi mirror is down

---
 tools/depends/target/flatbuffers/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/depends/target/flatbuffers/Makefile b/tools/depends/target/flatbuffers/Makefile
index fb30172000..652c13c72e 100644
--- a/tools/depends/target/flatbuffers/Makefile
+++ b/tools/depends/target/flatbuffers/Makefile
@@ -12,7 +12,7 @@ ifeq ($(PLATFORM),)
 	ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 	PLATFORM = native
 	TARBALLS_LOCATION = $(ROOT_DIR)
-	BASE_URL := http://mirrors.kodi.tv/build-deps/sources
+	BASE_URL := http://www.mirrorservice.org/sites/mirrors.kodi.tv/build-deps/sources
 	RETRIEVE_TOOL := curl -Ls --create-dirs -f -O
 	ARCHIVE_TOOL := tar --strip-components=1 -xf
 	CMAKE := cmake
-- 
2.16.4

