From 12010b6c65299366e958a218796afc5589293760 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Mon, 5 Jun 2017 16:36:47 +0100
Subject: [PATCH] Fix all-015

---
 xbmc/platform/posix/main.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/xbmc/platform/posix/main.cpp b/xbmc/platform/posix/main.cpp
index bee93da11a..e8b827988c 100644
--- a/xbmc/platform/posix/main.cpp
+++ b/xbmc/platform/posix/main.cpp
@@ -49,6 +49,17 @@
 #include "input/linux/LIRC.h"
 #endif
 
+#include "Application.h"
+#include "messaging/ApplicationMessenger.h"
+
+using namespace KODI::MESSAGING;
+
+void sigterm_handler(int signum)
+{
+       CLog::Log(LOGINFO, "OSMC: received signal to tear down Kodi");
+       if (! g_application.isShuttingDown())
+          CApplicationMessenger::GetInstance().PostMsg(TMSG_QUIT);
+}
 
 namespace
 {
-- 
2.12.3

