From 9ef42e94987b9d5dc6c6f2af5d11a29fb61cfc59 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Sat, 2 Jun 2018 07:23:50 +0100
Subject: [PATCH] test

---
 xbmc/AppParamParser.cpp | 20 +++++++++++++++++++-
 xbmc/AppParamParser.h   |  2 ++
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/xbmc/AppParamParser.cpp b/xbmc/AppParamParser.cpp
index 3729d39729..78d64b2ca7 100644
--- a/xbmc/AppParamParser.cpp
+++ b/xbmc/AppParamParser.cpp
@@ -25,6 +25,7 @@
 #include "utils/log.h"
 #include "utils/SystemInfo.h"
 #include "utils/StringUtils.h"
+#include "input/InputManager.h"
 #include <stdlib.h>

 CAppParamParser::CAppParamParser()
@@ -45,7 +46,24 @@ void CAppParamParser::Parse(const char* const* argv, int nArgs)
   if (nArgs > 1)
   {
     for (int i = 1; i < nArgs; i++)
-      ParseArg(argv[i]);
+      {
+	ParseArg(argv[i]);
+	if (strnicmp(argv[i], "-l", 2) == 0 || strnicmp(argv[i], "--lircdev", 9) == 0)
+        {
+         // check the next arg with the proper value.
+         int next = i + 1;
+         if (next < nArgs)
+         {
+           if ((argv[next][0] != '-') && (argv[next][0] == '/'))
+           {
+             //m_remoteControlName = argv[next];
+             i++;
+           }
+         }
+       }
+      // else if (strnicmp(argv[i], "-n", 2) == 0 || strnicmp(argv[i], "--nolirc", 8) == 0)
+      //   m_remoteControlEnabled = false;
+      }

    // testmode is only valid if at least one item to play was given
    if (m_playlist->IsEmpty())
diff --git a/xbmc/AppParamParser.h b/xbmc/AppParamParser.h
index df06e45d9b..1a6db3352b 100644
--- a/xbmc/AppParamParser.h
+++ b/xbmc/AppParamParser.h
@@ -27,6 +27,8 @@ class CAppParamParser
     void Parse(const char* const* argv, int nArgs);
     void SetAdvancedSettings(CAdvancedSettings& advancedSettings) const;
 
+    bool RemoteControlEnabled() const { return m_remoteControlEnabled; }
+    const std::string &RemoteControlName() const { return m_remoteControlName; }
     const CFileItemList& GetPlaylist() const;

     int m_logLevel;
@@ -38,4 +38,7 @@ class CAppParamParser

     std::string m_settingsFile;
     std::unique_ptr<CFileItemList> m_playlist;
+    bool m_remoteControlEnabled = true;
+    std::string m_remoteControlName;
+
 };
-- 
2.12.3

