From 2a48d0de75c35c3f596c783d96bf1110103b4e97 Mon Sep 17 00:00:00 2001
From: Kai Sommerfeld <kai.sommerfeld@gmx.com>
Date: Thu, 12 Apr 2018 21:13:17 +0200
Subject: [PATCH] [application] fix broken powermanagement after #13748.

---
 xbmc/Application.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 423c6c5983ff..bea858dc788a 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -586,15 +586,15 @@ bool CApplication::Create(const CAppParamParser &params)
   CWIN32Util::SetThreadLocalLocale(true); // enable independent locale for each thread, see https://connect.microsoft.com/VisualStudio/feedback/details/794122
 #endif // TARGET_WINDOWS
 
+  // audio (OSX) depends on WinSystem
+  m_pWinSystem = CWinSystemBase::CreateWinSystem();
+  CServiceBroker::RegisterWinSystem(m_pWinSystem.get());
+
   if (!m_ServiceManager->InitStageTwo(params))
   {
     return false;
   }

-  // audio (OSX) depends on WinSystem
-  m_pWinSystem = CWinSystemBase::CreateWinSystem();
-  CServiceBroker::RegisterWinSystem(m_pWinSystem.get());
-
   if (!m_ServiceManager->CreateAudioEngine())
   {
    CLog::Log(LOGFATAL, "CApplication::Create: Failed to load an AudioEngine");
