From 577e08236d434858b55ca68f4ffdd33d7c0cdae6 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Thu, 23 Nov 2017 14:39:35 +0000
Subject: [PATCH] Fixup all-090

---
 xbmc/storage/MediaManager.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/storage/MediaManager.cpp b/xbmc/storage/MediaManager.cpp
index 27a28cd68c..4b9333b064 100644
--- a/xbmc/storage/MediaManager.cpp
+++ b/xbmc/storage/MediaManager.cpp
@@ -681,6 +681,8 @@ std::vector<std::string> CMediaManager::GetDiskUsage()
 
 void CMediaManager::OnStorageAdded(const std::string &label, const std::string &path)
 {
+if (path== "/boot")
+  return;
 #ifdef HAS_DVD_DRIVE
   if (CServiceBroker::GetSettings().GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION) != AUTOCD_NONE || CServiceBroker::GetSettings().GetBool(CSettings::SETTING_DVDS_AUTORUN))
     if (CServiceBroker::GetSettings().GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION) == AUTOCD_RIP)
-- 
2.12.3

