From 6681b3ac95b0fd37a1ac58dcd8d6deec2748a208 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 23 Apr 2018 13:32:37 +0300
Subject: [PATCH] [networkservices] fix build without webserver

---
 xbmc/network/NetworkServices.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/xbmc/network/NetworkServices.cpp b/xbmc/network/NetworkServices.cpp
index 83f629ec3924..1ccbfbe00095 100644
--- a/xbmc/network/NetworkServices.cpp
+++ b/xbmc/network/NetworkServices.cpp
@@ -89,10 +89,9 @@ using namespace UPNP;
 using KODI::MESSAGING::HELPERS::DialogResponse;
 
 CNetworkServices::CNetworkServices(CSettings &settings)
+  : m_settings(settings)
 #ifdef HAS_WEB_SERVER
-  :
-  m_settings(settings),
-  m_webserver(*new CWebServer),
+  , m_webserver(*new CWebServer),
   m_httpImageHandler(*new CHTTPImageHandler),
   m_httpImageTransformationHandler(*new CHTTPImageTransformationHandler),
   m_httpVfsHandler(*new CHTTPVfsHandler),
