From 500430f8e7885d5e7dfce4fc8a07a1ecebe47f8f Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Sat, 16 Jun 2018 13:02:15 +0200
Subject: [PATCH] don't pass cryto to ffmpeg when transcoding

---
 .../VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp   | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp
index a3e7065462f3..6bc33c2efad2 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecAndroidMediaCodec.cpp
@@ -293,8 +293,11 @@ bool CDVDAudioCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
       }
       if (!m_decryptCodec)
       {
+        CDVDStreamInfo ffhints = hints;
+        ffhints.cryptoSession = nullptr;
+
         m_decryptCodec = std::shared_ptr<CDVDAudioCodec>(new CDVDAudioCodecFFmpeg(m_processInfo));
-        if (!m_decryptCodec->Open(hints, options))
+        if (!m_decryptCodec->Open(ffhints, options))
         {
           CLog::Log(LOGERROR, "CDVDAudioCodecAndroidMediaCodec::Open() Failed opening FFmpeg decoder");
           return false;
