From e699f6b5d7dd1e2680890b95471c746632ac8d03 Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Thu, 5 Jul 2018 14:01:38 +0200
Subject: [PATCH] guiinfo: grab gfx lock when registering GUIInfoProviders

---
 xbmc/GUIInfoManager.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/xbmc/GUIInfoManager.cpp b/xbmc/GUIInfoManager.cpp
index eab982f1c005..fd6af2888f4a 100644
--- a/xbmc/GUIInfoManager.cpp
+++ b/xbmc/GUIInfoManager.cpp
@@ -7059,10 +7059,20 @@ void CGUIInfoManager::OnApplicationMessage(KODI::MESSAGING::ThreadMessage* pMsg)
 
 void CGUIInfoManager::RegisterInfoProvider(IGUIInfoProvider *provider)
 {
+  if (!CServiceBroker::GetWinSystem())
+    return;
+
+  CSingleLock lock(CServiceBroker::GetWinSystem()->GetGfxContext());
+
   m_infoProviders.RegisterProvider(provider, false);
 }
 
 void CGUIInfoManager::UnregisterInfoProvider(IGUIInfoProvider *provider)
 {
+  if (!CServiceBroker::GetWinSystem())
+    return;
+
+  CSingleLock lock(CServiceBroker::GetWinSystem()->GetGfxContext());
+
   m_infoProviders.UnregisterProvider(provider);
 }
