From d3bc60cda5627f851814507c08a71c545992528e Mon Sep 17 00:00:00 2001
From: DaveTBlake <oak99sky@yahoo.co.uk>
Date: Thu, 3 Jan 2019 08:57:23 +0000
Subject: [PATCH] Coverity fix remove implicit fall through from switch

---
 xbmc/music/windows/MusicFileItemListModifier.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/xbmc/music/windows/MusicFileItemListModifier.cpp b/xbmc/music/windows/MusicFileItemListModifier.cpp
index 0768906f37b6..9187300751d8 100644
--- a/xbmc/music/windows/MusicFileItemListModifier.cpp
+++ b/xbmc/music/windows/MusicFileItemListModifier.cpp
@@ -55,18 +55,23 @@ void CMusicFileItemListModifier::AddQueuingFolder(CFileItemList& items)
   if (items.GetObjectCount() <= 1)
     return;
 
-  switch (directoryNode->GetChildType())
+  auto nodeChildType = directoryNode->GetChildType();
+
+  // No need for "all" when overview node and child node albums or artists
+  if (directoryNode->GetType() == NODE_TYPE_OVERVIEW &&
+     (nodeChildType == NODE_TYPE_ARTIST || nodeChildType == NODE_TYPE_ALBUM))
+    return;
+
+  switch (nodeChildType)
   {
   case NODE_TYPE_ARTIST:
-    if (directoryNode->GetType() == NODE_TYPE_OVERVIEW) return;
     pItem.reset(new CFileItem(g_localizeStrings.Get(15103)));  // "All Artists"
     musicUrl.AppendPath("-1/");
     pItem->SetPath(musicUrl.ToString());
     break;
 
-    //  All album related nodes
+  //  All album related nodes
   case NODE_TYPE_ALBUM:
-    if (directoryNode->GetType() == NODE_TYPE_OVERVIEW) return;
   case NODE_TYPE_ALBUM_RECENTLY_PLAYED:
   case NODE_TYPE_ALBUM_RECENTLY_ADDED:
   case NODE_TYPE_ALBUM_COMPILATIONS:
