From d85c4c892638111add99779cdee137ab2c9c6aeb Mon Sep 17 00:00:00 2001
From: "Hugh N. Greenberg" <hng@lanl.gov>
Date: Sun, 24 Feb 2019 22:02:50 -0700
Subject: [PATCH] Fixes a crash when a file's embedded art is invalid

---
 xbmc/video/tags/VideoTagLoaderNFO.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/video/tags/VideoTagLoaderNFO.cpp b/xbmc/video/tags/VideoTagLoaderNFO.cpp
index 16bbc0f26d96..1ca311f49ea8 100644
--- a/xbmc/video/tags/VideoTagLoaderNFO.cpp
+++ b/xbmc/video/tags/VideoTagLoaderNFO.cpp
@@ -40,10 +40,10 @@ CInfoScanner::INFO_TYPE CVideoTagLoaderNFO::Load(CVideoInfoTag& tag,
                                                  std::vector<EmbeddedArt>*)
 {
   CNfoFile nfoReader;
-  CInfoScanner::INFO_TYPE result;
-  if (m_info->Content() == CONTENT_TVSHOWS && !m_item.m_bIsFolder)
+  CInfoScanner::INFO_TYPE result = CInfoScanner::NO_NFO;
+  if (m_info && m_info->Content() == CONTENT_TVSHOWS && !m_item.m_bIsFolder)
     result = nfoReader.Create(m_path, m_info, m_item.GetVideoInfoTag()->m_iEpisode);
-  else
+  else if (m_info)
     result = nfoReader.Create(m_path, m_info);
 
   if (result == CInfoScanner::FULL_NFO || result == CInfoScanner::COMBINED_NFO)
