From c05f6298d5623a5613652be6fce6b1167c22e56b Mon Sep 17 00:00:00 2001
From: Rechi <Rechi@users.noreply.github.com>
Date: Wed, 27 Feb 2019 13:09:11 +0100
Subject: [PATCH] [fix] guard windows specific include

---
 xbmc/utils/test/TestSystemInfo.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/utils/test/TestSystemInfo.cpp b/xbmc/utils/test/TestSystemInfo.cpp
index 3692c97ed10a..ba715c173412 100644
--- a/xbmc/utils/test/TestSystemInfo.cpp
+++ b/xbmc/utils/test/TestSystemInfo.cpp
@@ -9,7 +9,9 @@
 #include "utils/SystemInfo.h"
 #include "settings/Settings.h"
 #include "GUIInfoManager.h"
+#if defined(TARGET_WINDOWS)
 #include "platform/win32/CharsetConverter.h"
+#endif
 
 #include "gtest/gtest.h"
 
