From 546ad2a43e0b724ca3950baa7db6524b221bb414 Mon Sep 17 00:00:00 2001
From: Kai Sommerfeld <kai.sommerfeld@gmx.com>
Date: Fri, 1 Mar 2019 11:16:54 +0100
Subject: [PATCH] [PVR] Fix crash on channel switch when no pvr client is
 enabled.

---
 xbmc/pvr/PVRGUIActions.cpp | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/xbmc/pvr/PVRGUIActions.cpp b/xbmc/pvr/PVRGUIActions.cpp
index 1ab35b086b50..240ebf8a06ca 100644
--- a/xbmc/pvr/PVRGUIActions.cpp
+++ b/xbmc/pvr/PVRGUIActions.cpp
@@ -1184,20 +1184,28 @@ namespace PVR
     switch (type)
     {
       case PlaybackTypeRadio:
+      {
         if (CServiceBroker::GetPVRManager().IsPlayingRadio())
           return true;
 
-        channel = CServiceBroker::GetPVRManager().ChannelGroups()->GetGroupAllRadio()->GetLastPlayedChannel();
+        const std::shared_ptr<CPVRChannelGroup> allGroup = CServiceBroker::GetPVRManager().ChannelGroups()->GetGroupAllRadio();
+        if (allGroup)
+          channel = allGroup->GetLastPlayedChannel();
+
         bIsRadio = true;
         break;
-
+      }
       case PlaybackTypeTV:
+      {
         if (CServiceBroker::GetPVRManager().IsPlayingTV())
           return true;
 
-        channel = CServiceBroker::GetPVRManager().ChannelGroups()->GetGroupAllTV()->GetLastPlayedChannel();
-        break;
+        const std::shared_ptr<CPVRChannelGroup> allGroup = CServiceBroker::GetPVRManager().ChannelGroups()->GetGroupAllTV();
+        if (allGroup)
+          channel = allGroup->GetLastPlayedChannel();
 
+        break;
+      }
       default:
         if (CServiceBroker::GetPVRManager().IsPlaying())
           return true;
