From 5e1cdc89f8a3d6f901f48a249a0200f094140465 Mon Sep 17 00:00:00 2001
From: arnova <arnova@void.org>
Date: Mon, 11 Feb 2019 21:14:15 +0100
Subject: [PATCH] fixed: NFO/tag loading for http/https sources was broken
 (fixes #15481)

---
 xbmc/video/tags/VideoInfoTagLoaderFactory.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/xbmc/video/tags/VideoInfoTagLoaderFactory.cpp b/xbmc/video/tags/VideoInfoTagLoaderFactory.cpp
index effc897c7dfa..dd6340b6e38d 100644
--- a/xbmc/video/tags/VideoInfoTagLoaderFactory.cpp
+++ b/xbmc/video/tags/VideoInfoTagLoaderFactory.cpp
@@ -22,10 +22,6 @@ IVideoInfoTagLoader* CVideoInfoTagLoaderFactory::CreateLoader(const CFileItem& i
                                                               bool lookInFolder,
                                                               bool forceRefresh)
 {
-  // don't try to read tags for streams
-  if (item.IsInternetStream())
-    return nullptr;
-
   if (item.IsPlugin() && info && info->ID() == "metadata.local")
   {
     // Direct loading from plugin source with metadata.local scraper
