From 0659fa3b67e162668803ba88b01994db9ad475b0 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 22 Jul 2018 14:49:17 +0200
Subject: [PATCH] GLContextEGL: Adjust sleeptime when detecting missed msc
 increase

---
 xbmc/windowing/X11/GLContextEGL.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/windowing/X11/GLContextEGL.cpp b/xbmc/windowing/X11/GLContextEGL.cpp
index ae84ea53f0cb..4f5075771504 100644
--- a/xbmc/windowing/X11/GLContextEGL.cpp
+++ b/xbmc/windowing/X11/GLContextEGL.cpp
@@ -374,12 +374,12 @@ void CGLContextEGL::SwapBuffers()
     // if no vertical retrace has occurred in eglSwapBuffers,
     // sleep until next vertical retrace
     uint64_t lastIncrement = (now / 1000 - ust2);
-    uint64_t sleeptime = m_sync.interval - lastIncrement;
     if (lastIncrement > m_sync.interval)
     {
       lastIncrement = m_sync.interval;
       CLog::Log(LOGWARNING, "CGLContextEGL::SwapBuffers: last msc time greater than interval");
     }
+    uint64_t sleeptime = m_sync.interval - lastIncrement;
     usleep(sleeptime);
     m_sync.cont++;
     msc2++;
@@ -389,12 +389,12 @@ void CGLContextEGL::SwapBuffers()
     // sleep until next vertical retrace
     // this avoids blocking outside of this function
     uint64_t lastIncrement = (now / 1000 - ust2);
-    uint64_t sleeptime = m_sync.interval - lastIncrement;
     if (lastIncrement > m_sync.interval)
     {
       lastIncrement = m_sync.interval;
       CLog::Log(LOGWARNING, "CGLContextEGL::SwapBuffers: last msc time greater than interval (1)");
     }
+    uint64_t sleeptime = m_sync.interval - lastIncrement;
     usleep(sleeptime);
     msc2++;
   }
