From 97e6a7f3d000520c1f8b68474c855ad4745d880b Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Thu, 7 Feb 2019 08:52:31 +0100
Subject: [PATCH] [DllLoader] No warning if resolve fals for optional
 dependencies

---
 xbmc/DynamicDll.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/DynamicDll.h b/xbmc/DynamicDll.h
index 3a7c535a3b81..b556a05183bc 100644
--- a/xbmc/DynamicDll.h
+++ b/xbmc/DynamicDll.h
@@ -381,11 +381,11 @@ public: \
 
 #define RESOLVE_METHOD_OPTIONAL(method) \
    m_##method##_ptr = nullptr; \
-   m_dll->ResolveExport( #method , & m_##method##_ptr );
+   m_dll->ResolveExport( #method , & m_##method##_ptr, false );
 
 #define RESOLVE_METHOD_OPTIONAL_FP(method) \
    method##_ptr = NULL; \
-   m_dll->ResolveExport( #method , & method##_ptr );
+   m_dll->ResolveExport( #method , & method##_ptr, false );
 
 
 
@@ -405,7 +405,7 @@ public: \
 
 #define RESOLVE_METHOD_RENAME_OPTIONAL(dllmethod, method) \
   m_##method##_ptr = nullptr; \
-  m_dll->ResolveExport( #dllmethod , & m_##method##_ptr );
+  m_dll->ResolveExport( #dllmethod , & m_##method##_ptr, false );
 
 #define RESOLVE_METHOD_RENAME_FP(dllmethod, method) \
   if (!m_dll->ResolveExport( #dllmethod , & method##_ptr )) \
