From 768906d6b0d18595a85e05dab4e6c7c7239e4caf Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Thu, 17 May 2018 19:10:05 +0200
Subject: [PATCH 1/2] [GUI] Invalidate underlying windows if modal dialog close

---
 xbmc/guilib/GUIWindowManager.cpp | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/xbmc/guilib/GUIWindowManager.cpp b/xbmc/guilib/GUIWindowManager.cpp
index af1bd3498f3d..80c5a82722b2 100644
--- a/xbmc/guilib/GUIWindowManager.cpp
+++ b/xbmc/guilib/GUIWindowManager.cpp
@@ -821,6 +821,8 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
     {
       CSingleExit exitit(CServiceBroker::GetWinSystem()->GetGfxContext());
       static_cast<CGUIDialog *>(pNewWindow)->Open(params.size() > 0 ? params[0] : "");
+      // Invalidate underlying windows after closing a modal dialog
+      MarkDirty();
     }
     return;
   }
@@ -1131,12 +1133,22 @@ void CGUIWindowManager::Process(unsigned int currentTime)
 
 void CGUIWindowManager::MarkDirty()
 {
-  m_tracker.MarkDirtyRegion(CDirtyRegion(CRect(0, 0, float(CServiceBroker::GetWinSystem()->GetGfxContext().GetWidth()), float(CServiceBroker::GetWinSystem()->GetGfxContext().GetHeight()))));
+  MarkDirty(CRect(0, 0, float(CServiceBroker::GetWinSystem()->GetGfxContext().GetWidth()), float(CServiceBroker::GetWinSystem()->GetGfxContext().GetHeight())));
 }
 
 void CGUIWindowManager::MarkDirty(const CRect& rect)
 {
   m_tracker.MarkDirtyRegion(CDirtyRegion(rect));
+
+  CGUIWindow* pWindow = GetWindow(GetActiveWindow());
+  if (pWindow)
+    pWindow->MarkDirtyRegion();
+
+  // make copy of vector as we may remove items from it as we go
+  auto activeDialogs = m_activeDialogs;
+  for (const auto& window : activeDialogs)
+    if (window->IsDialogRunning())
+      window->MarkDirtyRegion();
 }
 
 void CGUIWindowManager::RenderPass() const

From 8f2529d70a54ab8aa9a1ff991fa718f0041466b5 Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Sun, 20 May 2018 23:44:37 +0200
Subject: [PATCH 2/2] MarkRegionDirty if GUISpinControl changes value

---
 xbmc/guilib/GUISpinControl.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/guilib/GUISpinControl.cpp b/xbmc/guilib/GUISpinControl.cpp
index aeed306099dd..731d78add544 100644
--- a/xbmc/guilib/GUISpinControl.cpp
+++ b/xbmc/guilib/GUISpinControl.cpp
@@ -543,6 +543,9 @@ void CGUISpinControl::SetValueFromLabel(const std::string &label)
   }
   else
     m_iValue = atoi(label.c_str());
+
+  MarkDirtyRegion();
+  SetInvalid();
 }
 
 void CGUISpinControl::SetValue(int iValue)
@@ -557,6 +560,7 @@ void CGUISpinControl::SetValue(int iValue)
   else
     m_iValue = iValue;
 
+  MarkDirtyRegion();
   SetInvalid();
 }
 
