From 4ce18899a7395d0d497bdfef798e98a5398ddc1f Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Fri, 15 Jun 2018 18:44:01 +0200
Subject: [PATCH] VideoPlayer: only clear hints of streams on prepare

---
 xbmc/cores/VideoPlayer/VideoPlayer.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
index 892f62917ee0..c248a0d6d342 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
@@ -1260,11 +1260,11 @@ void CVideoPlayer::Prepare()
   m_processInfo->SetSpeed(1.0);
   m_processInfo->SetTempo(1.0);
   m_State.Clear();
-  m_CurrentVideo.Clear();
-  m_CurrentAudio.Clear();
-  m_CurrentSubtitle.Clear();
-  m_CurrentTeletext.Clear();
-  m_CurrentRadioRDS.Clear();
+  m_CurrentVideo.hint.Clear();
+  m_CurrentAudio.hint.Clear();
+  m_CurrentSubtitle.hint.Clear();
+  m_CurrentTeletext.hint.Clear();
+  m_CurrentRadioRDS.hint.Clear();
   memset(&m_SpeedState, 0, sizeof(m_SpeedState));
   m_offset_pts = 0;
   m_CurrentAudio.lastdts = DVD_NOPTS_VALUE;
