From acd50949e46428186e3cead423e572719eb41cd6 Mon Sep 17 00:00:00 2001
From: Bas Rieter <basrieter@gmail.com>
Date: Tue, 15 May 2018 17:04:15 +0200
Subject: [PATCH] Fixed: IsType check now uses m_strDynPath

---
 xbmc/FileItem.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/FileItem.cpp b/xbmc/FileItem.cpp
index a52c4b2fc1a8..4fc5bb08efec 100644
--- a/xbmc/FileItem.cpp
+++ b/xbmc/FileItem.cpp
@@ -1066,6 +1066,9 @@ bool CFileItem::IsPythonScript() const
 
 bool CFileItem::IsType(const char *ext) const
 {
+  if (!m_strDynPath.empty())
+    return URIUtils::HasExtension(m_strDynPath, ext);
+
   return URIUtils::HasExtension(m_strPath, ext);
 }
 
