From cc80622cfa539335f8a6e492eb9b12ff42b9f683 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Mon, 14 Jan 2019 20:54:47 +0100
Subject: [PATCH] FFmpeg: Fix missing include

---
 xbmc/cores/FFmpeg.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/cores/FFmpeg.h b/xbmc/cores/FFmpeg.h
index ffbe9422a4b2..b0e1a2ca1a4d 100644
--- a/xbmc/cores/FFmpeg.h
+++ b/xbmc/cores/FFmpeg.h
@@ -14,6 +14,7 @@ extern "C" {
 #include "libavcodec/avcodec.h"
 #include "libavformat/avformat.h"
 #include "libavutil/avutil.h"
+#include "libavutil/log.h"
 #include "libavutil/ffversion.h"
 #include "libavfilter/avfilter.h"
 #include "libpostproc/postprocess.h"
