From 1183eaeecd507390111c78abdc73d9dc38ac2ab4 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Thu, 28 Dec 2017 18:40:19 +0000
Subject: [PATCH] Register CAESinkALSA

---
 xbmc/windowing/rpi/WinSystemRpi.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/windowing/rpi/WinSystemRpi.cpp b/xbmc/windowing/rpi/WinSystemRpi.cpp
index 9e7f8bd9be..3e80952cee 100644
--- a/xbmc/windowing/rpi/WinSystemRpi.cpp
+++ b/xbmc/windowing/rpi/WinSystemRpi.cpp
@@ -35,6 +35,9 @@
 #include "../WinEventsLinux.h"
 #include "cores/AudioEngine/AESinkFactory.h"
 #include "cores/AudioEngine/Sinks/AESinkPi.h"
+// AESink Factory
+#include "cores/AudioEngine/AESinkFactory.h"
+#include "cores/AudioEngine/Sinks/AESinkALSA.h"
 
 #include <EGL/egl.h>
 #include <EGL/eglplatform.h>
@@ -55,6 +58,7 @@ CWinSystemRpi::CWinSystemRpi()
   m_winEvents.reset(new CWinEventsLinux());
   AE::CAESinkFactory::ClearSinks();
   CAESinkPi::Register();
+  CAESinkALSA::Register();
 }
 
 CWinSystemRpi::~CWinSystemRpi()
-- 
2.12.3

