From f0ed7df6ad311db4d72efbfbb1d8428a6c7ba6a8 Mon Sep 17 00:00:00 2001
From: peak3d <pfau@peak3d.de>
Date: Tue, 22 Jan 2019 16:35:19 +0100
Subject: [PATCH] [GUI] refresh debug overlay if smartredraw is activated

---
 xbmc/windows/GUIWindowDebugInfo.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/windows/GUIWindowDebugInfo.cpp b/xbmc/windows/GUIWindowDebugInfo.cpp
index 225fcb72fa11..2f0e6426e658 100644
--- a/xbmc/windows/GUIWindowDebugInfo.cpp
+++ b/xbmc/windows/GUIWindowDebugInfo.cpp
@@ -55,6 +55,9 @@ bool CGUIWindowDebugInfo::OnMessage(CGUIMessage &message)
     delete m_layout;
     m_layout = nullptr;
   }
+  else if (message.GetMessage() == GUI_MSG_REFRESH_TIMER)
+    MarkDirtyRegion();
+
   return CGUIDialog::OnMessage(message);
 }
 
