From 5b61165ecf2b630e00595bf0bdc2480aaa24a954 Mon Sep 17 00:00:00 2001
From: enen92 <92enen@gmail.com>
Date: Thu, 17 Jan 2019 16:04:35 +0000
Subject: [PATCH] [guilib] Ensure ListItems release the layout before it is
 destroyed

---
 xbmc/guilib/GUIBaseContainer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/guilib/GUIBaseContainer.cpp b/xbmc/guilib/GUIBaseContainer.cpp
index 7d1985badea8..59c7778cebf9 100644
--- a/xbmc/guilib/GUIBaseContainer.cpp
+++ b/xbmc/guilib/GUIBaseContainer.cpp
@@ -58,6 +58,10 @@ CGUIBaseContainer::CGUIBaseContainer(const CGUIBaseContainer &) = default;
 
 CGUIBaseContainer::~CGUIBaseContainer(void)
 {
+  // release the container from items
+  for (auto item :m_items)
+    item->FreeMemory();
+
   delete m_listProvider;
 }
 
