From 64660ba0709a9d4bda8ffa8641cfe1e46716901f Mon Sep 17 00:00:00 2001
From: wsnipex <wsnipex@a1.net>
Date: Mon, 21 Jan 2019 13:29:07 +0100
Subject: [PATCH] [StringUtils] fix crash on addon install when using fmt 5.3

---
 xbmc/utils/StringUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/utils/StringUtils.cpp b/xbmc/utils/StringUtils.cpp
index c08b94e3e184..79fc0a75c8d5 100644
--- a/xbmc/utils/StringUtils.cpp
+++ b/xbmc/utils/StringUtils.cpp
@@ -1265,7 +1265,7 @@ std::string StringUtils::FormatFileSize(uint64_t bytes)
     value /= 1024.0;
   }
   unsigned int decimals = value < 9.995 ? 2 : (value < 99.95 ? 1 : 0);
-  auto frmt = "%.0" + Format("%u", decimals) + "f%s";
+  auto frmt = "%." + Format("%u", decimals) + "f%s";
   return Format(frmt.c_str(), value, units[i].c_str());
 }
 
