From bc9cd518cb153224c6cc31c7326dc77347875024 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Tue, 6 Jun 2017 08:37:40 +0100
Subject: [PATCH] Dirt hack for fmt

---
 cmake/modules/FindFmt.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/modules/FindFmt.cmake b/cmake/modules/FindFmt.cmake
index 5473ed5b76..d7205e287c 100644
--- a/cmake/modules/FindFmt.cmake
+++ b/cmake/modules/FindFmt.cmake
@@ -12,12 +12,12 @@
 #
 #   Fmt::Fmt   - The Fmt library
 
-if(CORE_SYSTEM_NAME STREQUAL windows OR CORE_SYSTEM_NAME STREQUAL windowsstore)
+#if(CORE_SYSTEM_NAME STREQUAL windows OR CORE_SYSTEM_NAME STREQUAL windowsstore)
   # TODO: fix windows fmt package to include fmt-config.cmake and fmt-config-version.cmake
   set(FMT_VERSION 3.0.1)
-else()
-  find_package(FMT 3.0.1 CONFIG REQUIRED QUIET)
-endif()
+#else()
+#  find_package(FMT 3.0.1 CONFIG REQUIRED QUIET)
+#endif()
 
 if(PKG_CONFIG_FOUND)
   pkg_check_modules(PC_FMT libfmt QUIET)
-- 
2.12.3

