From 85fcac7315bf4340b0a28a4eac9c623622deacbc Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Sun, 28 May 2017 12:20:28 +0100
Subject: [PATCH] Add deps to cmake

---
 cmake/platform/linux/aml.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/platform/linux/aml.cmake b/cmake/platform/linux/aml.cmake
index c1d464d2d2..bfeecf4a20 100644
--- a/cmake/platform/linux/aml.cmake
+++ b/cmake/platform/linux/aml.cmake
@@ -1 +1,2 @@
 set(PLATFORM_REQUIRED_DEPS OpenGLES AML EGL)
+set(PLATFORM_OPTIONAL_DEPS CEC Bluray AML NFS)
-- 
2.12.0

