From eda5c0e96060bb1a826a53b9eae0e718ae7352a2 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Fri, 9 Jun 2017 07:58:53 +0100
Subject: [PATCH] Add RPATH to cmake

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfe4091085..11bbd4badf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,8 @@ endif()
 # Set CORE_BUILD_DIR
 set(CORE_BUILD_DIR build)
 
+SET(CMAKE_INSTALL_RPATH "/usr/osmc/lib")
+
 message(STATUS "Source directory: ${CMAKE_SOURCE_DIR}")
 message(STATUS "Build directory: ${CMAKE_BINARY_DIR}")
 if(CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
-- 
2.12.3

