From fdc3c4a17c6de78bd5afc9c49daaf197a13be8e2 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Fri, 12 Oct 2018 06:40:21 +0100
Subject: [PATCH] Update vero3-098

---
 xbmc/Application.cpp | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index be61e6c9b2..5488183574 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2540,25 +2540,6 @@ void CApplication::Stop(int exitCode)
 
     g_alarmClock.StopThread();
 
-    CLog::Log(LOGNOTICE, "Storing total System Uptime");
-    g_sysinfo.SetTotalUptime(g_sysinfo.GetTotalUptime() + (int)(CTimeUtils::GetFrameTime() / 60000));
-
-    // Update the settings information (volume, uptime etc. need saving)
-    if (CFile::Exists(CServiceBroker::GetSettingsComponent()->GetProfileManager()->GetSettingsFile()))
-    {
-      CLog::Log(LOGNOTICE, "Saving settings");
-      CServiceBroker::GetSettingsComponent()->GetSettings()->Save();
-    }
-    else
-      CLog::Log(LOGNOTICE, "Not saving settings (settings.xml is not present)");
-
-    // kodi may crash or deadlock during exit (shutdown / reboot) due to
-    // either a bug in core or misbehaving addons. so try saving
-    // skin settings early
-    CLog::Log(LOGNOTICE, "Saving skin settings");
-    if (g_SkinInfo != nullptr)
-      g_SkinInfo->SaveSettings();
-
     m_bStop = true;
     // Add this here to keep the same ordering behaviour for now
     // Needs cleaning up
-- 
2.16.4

