From 891e58347652b3575c751494579a7d2fb1ebde71 Mon Sep 17 00:00:00 2001
From: afl1 <afl2001@gmail.com>
Date: Fri, 11 May 2018 13:34:03 +0200
Subject: [PATCH] Application: register Stereoscopicmanager for message target

---
 xbmc/Application.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 47c85a3451a3..a6ae2cb179c3 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -1612,6 +1612,7 @@ bool CApplication::LoadSkin(const std::string& skinID)
   CServiceBroker::GetGUI()->GetWindowManager().AddMsgTarget(this);
   CServiceBroker::GetGUI()->GetWindowManager().AddMsgTarget(&CServiceBroker::GetPlaylistPlayer());
   CServiceBroker::GetGUI()->GetWindowManager().AddMsgTarget(&g_fontManager);
+  CServiceBroker::GetGUI()->GetWindowManager().AddMsgTarget(&CServiceBroker::GetGUI()->GetStereoscopicsManager());
   CServiceBroker::GetGUI()->GetWindowManager().SetCallback(*this);
   //@todo should be done by GUIComponents
   CServiceBroker::GetGUI()->GetWindowManager().Initialize();
