From bcbe1a6546b303669c6c5fb7ac4da1ea71af7a1d Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Sun, 31 Dec 2017 12:03:17 +0000
Subject: [PATCH] Update patch all-094

---
 xbmc/Application.cpp | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 627402d48c..5c401219d7 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2822,25 +2822,6 @@ void CApplication::Stop(int exitCode)
 
     g_alarmClock.StopThread();
 
-    CLog::Log(LOGNOTICE, "Storing total System Uptime");
-    g_sysinfo.SetTotalUptime(g_sysinfo.GetTotalUptime() + (int)(CTimeUtils::GetFrameTime() / 60000));
-
-    // Update the settings information (volume, uptime etc. need saving)
-    if (CFile::Exists(m_ServiceManager->GetProfileManager().GetSettingsFile())) 
-    {
-      CLog::Log(LOGNOTICE, "Saving settings");
-      m_ServiceManager->GetSettings().Save();
-    }
-    else
-      CLog::Log(LOGNOTICE, "Not saving settings (settings.xml is not present)");
-
-    // kodi may crash or deadlock during exit (shutdown / reboot) due to
-    // either a bug in core or misbehaving addons. so try saving
-    // skin settings early
-    CLog::Log(LOGNOTICE, "Saving skin settings");
-    if (g_SkinInfo != nullptr)
-      g_SkinInfo->SaveSettings();
-
     m_bStop = true;
     // Add this here to keep the same ordering behaviour for now
     // Needs cleaning up
-- 
2.12.3

