From 475783c6776582d834ddfc4cd1a37d1b3273e5b9 Mon Sep 17 00:00:00 2001
From: ace20022 <ace20022@ymail.com>
Date: Wed, 22 Aug 2018 15:41:26 +0200
Subject: [PATCH] [windowing][vp] Fix mouse selection in disc menus.

---
 xbmc/video/windows/GUIWindowFullScreen.cpp | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/xbmc/video/windows/GUIWindowFullScreen.cpp b/xbmc/video/windows/GUIWindowFullScreen.cpp
index f6876bbf45d9..b31d70c33def 100644
--- a/xbmc/video/windows/GUIWindowFullScreen.cpp
+++ b/xbmc/video/windows/GUIWindowFullScreen.cpp
@@ -95,14 +95,21 @@ bool CGUIWindowFullScreen::OnAction(const CAction &action)
   case ACTION_MOUSE_MOVE:
     if (action.GetAmount(2) || action.GetAmount(3))
     {
-      TriggerOSD();
-      return true;
+      if (!g_application.GetAppPlayer().IsInMenu())
+      {
+        TriggerOSD();
+        return true;
+      }
     }
     break;
 
   case ACTION_MOUSE_LEFT_CLICK:
-    TriggerOSD();
-    return true;
+    if (!g_application.GetAppPlayer().IsInMenu())
+    {
+      TriggerOSD();
+      return true;
+    }
+    break;
 
   case ACTION_SHOW_GUI:
     {
